crates/ruma-common/src/events/message.rs

```diff
@@ -128,6 +128,17 @@ impl MessageEventContent {
 pub struct MessageContent(pub(crate) Vec<Text>);
 
 impl MessageContent {
+    /// Create a `MessageContent` from an array of messages.
+    ///
+    /// Returns `None` if the array is empty.
+    pub fn from(messages: Vec<Text>) -> Option<Self> {
```

Comment:
Having a constructor that's just named `from` is pretty confusing because `Type::from` usually means `<Type as From<_>>::from`. Let's name this `from_vec` or maybe even just `new`.

Also the `TryFrom` impl can be implemented by calling this + `ok_or`.